### PR TITLE
Feature/joi ui upgrade

### PR DESCRIPTION
**Pull request by hkernbach**
This PR is upgrading the Joi library to the newest version.
Also it takes care of a smaller regex issue: https://github.com/arangodb/planning/issues/3142

Before review, please re-build the UI: `make frontend` - Please review (carefully). 
Review: Do mop tests (click click) in other related views which are also using Joi.

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/frontend/js/arango/arango.js [hunk 1]
```diff
@@ -1233,8 +1233,8 @@
           validateInput: function () {
             return [
               {
-                rule: Joi.string().regex(/^(\/(APP[^\/]+|(?!APP)[a-zA-Z0-9_\-\/]+))+$/i),
-                msg: 'May not contain /APP'
+                rule: Joi.string().regex(/^(.(?!APP(\/|$)/i, {invert: true}),
```
Comment (pluma) [inline js/apps/system/_admin/aardvark/APP/frontend/js/arango/arango.js]: Unless I'm mistaken this is a syntax error (count the parens).
Also the leading paren is unnecessary.
Also this regex doesn't seem to work?
Any reason you don't simply split by `/` and check if any of the items in the array matches "APP"?

Comment (hkernbach) [reply js/apps/system/_admin/aardvark/APP/frontend/js/arango/arango.js]: ups, let me check! Yes that looks wrong.

Comment (hkernbach) [reply js/apps/system/_admin/aardvark/APP/frontend/js/arango/arango.js]: > Any reason you don't simply split by / and check if any of the items in the array matches "APP"?

That would also be possible, but I thought we want to stay to the regex stuff.

**Comment by pluma** [comment]
Can you link the related PR for the other version(s)?

**Comment by hkernbach** [comment]
https://github.com/arangodb/arangodb/pull/7322

**Comment by hkernbach** [comment]
and https://github.com/arangodb/arangodb/pull/7321

**Pull request merged**